lib/features/qr/ui/qr_page.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:sf_qr_code/features/qr/bloc/qr_bloc.dart';
+import 'package:sf_qr_code/themes/app_colors.dart';
+
+class QRPage extends StatefulWidget {
+  @override
+  _QRPageState createState() => _QRPageState();
+}
+
+class _QRPageState extends State<QRPage> {
+  String displayTimer = "";
+  num timerNum = 0;
+  Timer? timer;
+  int _start = 0; // default
+
+  @override
+  void initState() {
+    BlocProvider.of<QrBloc>(context).add(QrSeedRequestEvent());
+    setState(() {
+      displayTimer = "";
+      timerNum = 0;
+    });
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<QrBloc, QrState>(
```

Comment:
**question:** do you see any performance issues with this approach specifically related to the widget tree?